packages/ecc-client-ga4gh-tes/src/components/tes-service/definition/tesService.template.ts

```diff
@@ -0,0 +1,107 @@
+import { html, repeat, when } from '@microsoft/fast-element';
+
+const OtherTemplate: any = (x: any) => html`
+  <div class="template-container container key-value">
+    <div class="key">${x[0]}</div>
+    <div class="value">${x[1]}</div>
+  </div>
+`;
+
+const ArrayTemplate: any = (x: any) => html`
+  <div class="template-container">
+    <div class="container array-container">
+      <div class="key">${x[0]}</div>
+      <div class="value array-value">
+        ${repeat(
+          (arr) => arr[1],
+          html`
+            ${when(
+              (val) => Array.isArray(val[1]),
+              html`${(val) => ArrayTemplate(val)} `
+            )}
+            ${when(
+              (val) =>
+                typeof val[1] === 'object' &&
+                val[1] !== null &&
+                !Array.isArray(val[1]),
+              html` ${(val) => ObjectTemplate(val)} `
+            )}
+            ${when(
+              (val) => typeof val[1] !== 'object',
+              html` ${(val) => val} `
+            )}
+          `
+        )}
+      </div>
+    </div>
+  </div>
+`;
+
+const ObjectTemplate: any = (x: any) => html`
+  <div class="template-container">
+    <div class="obj-name">${x[0]}:</div>
+    <div class="object-container">
+      <div class="value object-value">
+        ${when(
+          (obj) => Object.entries(obj[1]).length > 0,
+          html`
+            ${repeat(
+              (val) => Object.entries(val[1]),
+              html`
+                ${when(
+                  (val) => Array.isArray(val[1]),
+                  html` ${(val) => ArrayTemplate(val)} `
+                )}
+                ${when(
+                  (val) =>
+                    typeof val[1] === 'object' &&
+                    val[1] !== null &&
+                    !Array.isArray(x[1]),
+                  html` ${(val) => ObjectTemplate(val)} `
+                )}
+                ${when(
+                  (val) => typeof val[1] !== 'object',
+                  html` ${(val) => OtherTemplate(val)} `
```

Comment:
Maybe, we can combine this repeated logic!

Reply:
Since only the bottom two have duplicate code, but one is a function other is a variable, trying to combine it is not worth it as it needs the context (x) as well? or am I missing something.

Reply:
I guess if the array implementation is changed we can combine them. Not sure though also not a big conflict for me!

Reply:
Done, please check and resolve.